launcher/ui/MainWindow.cpp

```diff
@@ -1279,7 +1279,17 @@ void MainWindow::globalSettingsClosed()
 
 void MainWindow::on_actionEditInstance_triggered()
 {
-    APPLICATION->showInstanceWindow(m_selectedInstance);
+
+    if (!m_selectedInstance)
+        return;
+
+    if (m_selectedInstance->canEdit()) {
+        APPLICATION->showInstanceWindow(m_selectedInstance);
+    } else  {
+        CustomMessageBox::selectable(this, tr("Instance not editable"), 
+                                     tr("This instance is not editable. It may be broken, invalid, or too old. Check logs for details."),
+                                     QMessageBox::Critical)->show();
```

Comment:
Wait, shouldn't this be `exec`? Oops...

Reply:
why would it need to be `exec`? `exec` creates it;s own event loop and pauses execution at the line until the windows closes, it's also discouraged as indiscriminate use can lead to memory safety issues.
we don't need that here so we can just show the window.

Reply:
show should be fine

Reply:
> why would it need to be `exec`? `exec` creates it;s own event loop and pauses execution at the line until the windows closes, it's also discouraged as indiscriminate use can lead to memory safety issues. we also don't need the return value so no need to use `open` we don't need that here so we can just `show` the window.

Will it be a modal?

Reply:
I think I missed the giant warning on the method lol.

if `show` doesn't make the dialog modal, i think `show` would be best